doc/man3/ASYNC_WAIT_CTX_new.pod

```diff
@@ -103,14 +120,47 @@ code that the job should be resumed by making the wait file descriptor
 "readable". Once resumed the engine should clear the wake signal on the wait
 file descriptor.
 
+Besides file descriptor, an optional additional method of notification is
+provided via callback owned by user code. The callback and data pointers would
+be stored within the ASYNC_WAIT_CTX along with an additional status field that
+can be used for the notification of retries from an engine. This additional
+method can be used when user thinks the file descriptor is too costly in terms
+of CPU cycles or in some context where file descriptor is not appropriate.
+
+An example of the usage of this additional method might be the following.
+User code would initiate cryptographic operations, and engine code would dispatch
+this operation to hardware, and if the dispatch is successful, then engine code
+would call ASYNC_pause_job() to return control to the user code. After that,
+user code can perform other tasks. When hardware completes the operation,
+normally it is detected by a polling function or an interrupt, as user code set
+callback by calling ASYNC_WAIT_CTX_set_callback() previously, then the
+registered callback will be called which is normally to resume the paused
+working flow. "status" is used when there is something abnormal happened when
+engine code dispatches the cryptographic operation to hardware, and
+ASYNC_WAIT_CTX_set_status() is called to pass information from engine to user
+code, so user code can base on the status to perform the further action.
+ASYNC_STATUS_ERR indicates HW has fatal problem, thus the crypto request is
+not submitted to HW, and suggest user code to stop this session.
+ASYNC_STATUS_OK means the request is successfully sent to HW, and
+ASYNC_STATUS_EAGAIN means the engine has some problem which can be recovered
+soon, such as buffer is full, so please send it next time.  "status" can also be
+used to tell if this additional callback method is implemented in engine or not,
+and if the engine does not implement it, "status" field would not be touched, so
+ASYNC_STATUS_UNSUPPORTED will be return via ASYNC_WAIT_CTX_get_status().
+Based on this, user code can be written independently of any specific engine to
+support either file descriptors or callbacks method.
```

Comment:
This needs rewriting. Please describe each of the functions and their arguments first. When describing ASYNC_WAIT_CTX_set_status() or ASYNC_WAIT_CTX_get_status() list each of the possible status values and what they mean (e.g. perhaps using pod "item" lists). You can include a usage example at the end.